libs/wingsdk/src/target-sim/redis.inflight.ts

```diff
@@ -0,0 +1,79 @@
+import Dockerode from "dockerode";
+import { ISimulatorResourceInstance } from "./resource";
+import { RedisSchema } from "./schema-resources";
+import { IRedisClient } from "../redis";
+import { ISimulatorContext } from "../testing/simulator";
+
+// Issue using types from ioredis with JSII
+//import { Redis as IoRedis } from "ioredis";
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+const IoRedis = require("ioredis");
+
+export class Redis implements IRedisClient, ISimulatorResourceInstance {
+  static uid = 0;
+  private readonly container_name: string;
+  private readonly context: ISimulatorContext;
+
+  private connection_url?: string = undefined;
+  private docker?: Dockerode = undefined;
+  private connection?: any;
+
+  public constructor(_props: RedisSchema["props"], context: ISimulatorContext) {
+    this.context = context;
+    this.container_name = `wing-sim-redis-${this.context.resourcePath.replace(
+      "/",
+      "."
+    )}-${Redis.uid}`;
```

Comment:
Perhaps use a `uuid()`. Generally trying to avoid static mutable state.

Reply:
This is just a sim hack and I didn't want to add more dependencies here and also make the docker container name very long. Make sense?

Reply:
@hasanaburayyan maybe check this static field and just use the generated port number in the name that that we've got that.

Reply:
Yupp, Ill remove the uid entirely using the port instead

Reply:
I ran into issues with portfinder, I opted for a different solution using port `0` wildcard and then inspecting running container to retrieve port (for url). This means I cannot provide the port number in the container name. So I used `uuid()`